bundles/persistence/org.openhab.persistence.jpa/java/org/openhab/persistence/jpa/internal/JpaPersistenceService.java

```diff
@@ -211,8 +211,12 @@ protected EntityManagerFactory newEntityManagerFactory() {
 		Map<String, String> properties = new HashMap<String, String>();
 		properties.put("javax.persistence.jdbc.url", JpaConfiguration.dbConnectionUrl);
 		properties.put("javax.persistence.jdbc.driver", JpaConfiguration.dbDriverClass);
-		properties.put("javax.persistence.jdbc.user", JpaConfiguration.dbUserName);
-		properties.put("javax.persistence.jdbc.password", JpaConfiguration.dbPassword);
+		if(JpaConfiguration.dbUserName != null) {
+		    properties.put("javax.persistence.jdbc.user", JpaConfiguration.dbUserName);
+		}
+		if(JpaConfiguration.dbPassword != null) {
+		    properties.put("javax.persistence.jdbc.password", JpaConfiguration.dbPassword);
+		}
```

Comment:
probably a good idea to INFO/WARN log a message stating that we strongly advice to set a username + password?